File: Frogger/roadMargin.cpp
#include "roadMargin.h"

RoadMargin::RoadMargin(float x, float y, float z) : StaticObject(x, y, z) {

	cube = new Cube();
	setupObjects();
}

void RoadMargin::setupObjects() {

	float road_margin_amb[] = { 0.0f, 0.0f, 0.0f, 1.0f };
	float road_margin_diff[] = { 0.55f, 0.55f, 0.55f, 1.0f };
	float road_margin_spec[] = { 0.70f, 0.70f, 0.70f, 1.0f };
	float road_margin_shin[] = { 80.0f };

	//River
	cube->setColor(VSResourceLib::AMBIENT, road_margin_amb);
	cube->setColor(VSResourceLib::DIFFUSE, road_margin_diff);
	cube->setColor(VSResourceLib::SPECULAR, road_margin_spec);
	cube->setColor(VSResourceLib::SHININESS, road_margin_shin);
}

void RoadMargin::render(VSShaderLib shader) {

	vsml->loadIdentity(VSMathLib::MODEL);

	vsml->pushMatrix(VSMathLib::MODEL);
	vsml->translate(position[0], position[1], position[2]);
	vsml->scale(BORDER_SCALE_FACTOR_X, 1, SCALE_FACTOR_Z);
	cube->render(shader);
	vsml->popMatrix(VSMathLib::MODEL);
}

File: FroggerJS/roadMargin.js
function RoadMargin(x, y, z) {

	this.x = x;
	this.y = y;
	this.z = z;

	this.cube = new Cube();
	this.light = new Cube();
	this.setupObjects();
}


RoadMargin.prototype.setupObjects = function() {

	this.cube.setColor("ambient", [0.25, 0.25, 0.25, 1.0]);
	this.cube.setColor("diffuse", [0.55, 0.55, 0.55, 1.0]);
	this.cube.setColor("specular", [0.70, 0.70, 0.70, 1.0]);
	this.cube.setColor("emissive", [0.0, 0.0, 0.0, 1.0]);
	this.cube.setColor("shininess", 80.0);
	
	this.light.setColor("ambient", [0.19225, 0.19225, 0.19225, 1.0 ]);
	this.light.setColor("diffuse", [0.50754, 0.50754, 0.50754, 1.0]);
	this.light.setColor("specular", [.508273, 0.508273, 0.508273, 1.0]);
	this.light.setColor("emissive", [0.0, 0.0, 0.0, 1.0]);
	this.light.setColor("shininess", 51.2);

}

RoadMargin.prototype.render = function() {

	mat4.identity(mMatrix);

	pushMatrix();

	mat4.translate(mMatrix, [this.x, this.y, this.z]);
	mat4.scale(mMatrix, [4.0, 1.0, 100.0]);
	this.cube.render(shaderProgram);
	
	popMatrix();

	//Lights 
	/*
	pushMatrix();
	
	mat4.translate(mMatrix, [2.0, 3.0, 4.0]);
	mat4.scale(mMatrix, [1.0, 6.0, 1.0]);
	this.light.render(shaderProgram);
	
	popMatrix();
	
	pushMatrix();
	
	mat4.translate(mMatrix, [3.5, 6.0, 4.0]);
	mat4.scale(mMatrix, [4.0, 1.0, 1.0]);
	this.light.render(shaderProgram);
	
	popMatrix();

	pushMatrix();
	
	mat4.translate(mMatrix, [2.0, 3.0, 36.0]);
	mat4.scale(mMatrix, [1.0, 6.0, 1.0]);
	this.light.render(shaderProgram);
	
	popMatrix();
	
	pushMatrix();
	
	mat4.translate(mMatrix, [3.5, 6.0, 36.0]);
	mat4.scale(mMatrix, [4.0, 1.0, 1.0]);
	this.light.render(shaderProgram);
	
	popMatrix();

	pushMatrix();
	
	mat4.translate(mMatrix, [2.0, 3.0, 76.0]);
	mat4.scale(mMatrix, [1.0, 6.0, 1.0]);
	this.light.render(shaderProgram);
	
	popMatrix();
	
	pushMatrix();
	
	mat4.translate(mMatrix, [3.5, 6.0, 76.0]);
	mat4.scale(mMatrix, [4.0, 1.0, 1.0]);
	this.light.render(shaderProgram);
	
	popMatrix();*/
}